IPIP/0000-double-hash-dht.md

```diff
@@ -0,0 +1,333 @@
+
+However Prefix Requests don't offer [`l`-diversity](https://en.wikipedia.org/wiki/L-diversity) nor [`t`-closeness](https://en.wikipedia.org/wiki/T-closeness), as frequency analysis is still easy to perform. For example, a `Prefix` matches a very popular Provider Records and a few unpopular ones. The DHT Server nodes can take a better-than-random guess when a new request is received for this `Prefix` that there is a higher chance that the Client is requesting the popular content's Provider Record compared with an unpopular one. However, the DHT Server cannot prove the the Client has accessed the popular content.
+
+**Provider Record Encryption**
+
+Provider Record Encryption also builds on top of Double Hashing. The Provider Record Encryption prevents curious DHT Servers observing a request for `Prefix` but not storing any Provider Record matching `Prefix`, to replay the request for `Prefix` and get all published keys matching `Prefix` including `HASH2` of the content accessed by the Client. It prevents all curious actors from building a global dictionary of `HASH2` to Content Providers for all content published in the IPFS public DHT. It is necessary to know the `MH` of the content (included in the `CID`) to learn about its Content Providers.
+
+Curious DHT Servers observing a request from `PeerID` for `Prefix` cannot associate the Client's `PeerID` with the Content Provider's `PeerID` because they cannot read the Provider Record.
+
+### Writer Privacy
+
+Writer Privacy is NOT the goal of this design. However, as a side effect, Write Privacy gets improved in some specific cases.
+- Content Providers do NOT get any additional privacy from the Client fetching the data
+- Content Providers can now hide to the DHT Server peers hosting their Provider Records which data they are serving, as long as the DHT Servers don't know the preimage of `HASH2`: `MH`. The DHT Servers are not able to query the content associated with the Provider Records they are storing. However, they can approximately monitor the number of requests associated to the content by observing the requests in the keysubspace matching to `Prefix` of `HASH2` of the content. DHT Servers can take an educated guess on the association of `HASH2` with the Content Provider's `PeerID`. The DHT Servers storing the Provider Record are able to share information about the Content Provider with potential accomplices.
+- Content Providers get additional privacy from curious DHT Servers observing a request, but NOT storing the Provider Record. These DHT Servers can still replay the DHT request, but are unable to discover the Content Provider's `PeerID` associated with `Preix` because the Provider Records are encrypted, and the content itself. This holds as long as the DHT Servers don't know the `MH` (or `CID`).
```

Comment:
Yes, the content is encrypted by default on libp2p. However, the Provider Records are encrypted using `MH`. So if a Client makes a DHT request, a curious intermediary DHT Server (not storing the Provider Record) will learn `Prefix`. The curious intermediary DHT Server can then make a request for `Prefix`, which will eventually open an encrypted libp2p stream with a DHT Server storing matching Provider Records. This encrypted libp2p stream is used to send the Provider Records matching `Prefix` back to the curious intermediary server. 
If there is no additional encryption (on top of the libp2p stream) the curious intermediary DHT Server can learn the Content Provider's `PeerID`. If the Provider Record is encrypted using `MH`, the curious intermediary DHT Server cannot learn anything from it.

I slightly rephrase to add clarity:

These DHT Servers can still replay the DHT request, but are unable to discover the content and the Content Provider's `PeerID` associated with `Prefix`, because the Provider Records are encrypted using `MH`.